### PR TITLE
Fix colormap keyword argument handling

### PR DESCRIPTION
**Pull request by lassepe**
I just did the minimum viable fix. I think that the way in which keyword arguments are handled here is very error-prone. Maybe the constants in `maps_data.jl` should be `NamedTuples` of `NamedTuples` to avoid manual indexing at various places.

### CODE REVIEW

File: src/colormaps.jl [hunk 1]
```diff
@@ -284,31 +284,35 @@ function colormap(cname::AbstractString, N::Int=100; mid=0.5, logscale=false, kv
...
+        allowedkeys = [:h, :w, :d, :c, :s, :b, :wcolor, :dcolor]
         p = copy(colormaps_sequential[cname][1:8])
 
-        allowedkeys = [:h, :w, :d, :c, :s, :b, :wcolor, :dcolor]
         for (k,v) in kvs
-            k in allowedkeys || throw(ArgumentError("Unknown keyword argument $k"))
-            ind = findfirst(allowedkeys, k)
+            ind = findfirst(e->e==k, allowedkeys)
+            ind != nothing || throw(ArgumentError("Unknown keyword argument $k"))
             if ind > 0
```
[the remark this answers is not in the document]
Comment (lassepe) [reply src/colormaps.jl]: Yes they never were. I rather left them for historical reasons. Will fix.

File: src/colormaps.jl [hunk 2]
```diff
@@ -284,31 +284,35 @@ function colormap(cname::AbstractString, N::Int=100; mid=0.5, logscale=false, kv
...
     if haskey(colormaps_sequential, cname)
+        allowedkeys = [:h, :w, :d, :c, :s, :b, :wcolor, :dcolor]
         p = copy(colormaps_sequential[cname][1:8])
 
-        allowedkeys = [:h, :w, :d, :c, :s, :b, :wcolor, :dcolor]
         for (k,v) in kvs
-            k in allowedkeys || throw(ArgumentError("Unknown keyword argument $k"))
-            ind = findfirst(allowedkeys, k)
+            ind = findfirst(e->e==k, allowedkeys)
+            ind != nothing || throw(ArgumentError("Unknown keyword argument $k"))
```
[the remark this answers is not in the document]
Comment (lassepe) [reply src/colormaps.jl]: I don't have a strong preference here. I just rather see these as assertions, which is why I write things with `||`.

**Comment by lassepe** [comment]
> I totally agree. I wonder how many users put this feature to good use.

Looking at the history, this must have been broken for more than a year. Given the fact, that there was no issue regarding this, I assume barely anyone is using this.

File: test/colormaps.jl [hunk 3]
```diff
@@ -13,13 +13,40 @@
...
 
     @test isconcretetype(eltype(colormap("Grays")))
 
     @test_throws ArgumentError colormap("Grays", N=10) # optional arguments, not keyword
 
+    # not return values to check here, just checking that keywords can be used
+    # Sequential
+    colormap("Blues", 10; mid=0.5)
+    colormap("Blues", 10; logscale=true)
+    colormap("Blues", 10; h=0.5)
```
[the remark this answers is not in the document]
Comment (lassepe) [reply test/colormaps.jl]: Fixed

File: src/colormaps.jl (hunk 2 above)
[the remark this answers is not in the document]
Comment (kimikage) [reply src/colormaps.jl]: I understand. I don't have a strong preference, too. I saw these as simple `if`-`then` statements since `kvs` are user inputs.

**Comment by cormullion** [comment]
@kimikage I tend to work mainly on documentation-related aspects, leaving the technicalities of colour science to others. I'd say that a minor maintenance release every month or so is a good basis, if some improvements and fixes have accumulated. As long as tests continue to improve as well, I'm happy to see more committers!

**Comment by kimikage** [comment]
Thank you for your comments. Since the last release(v0.9.6), it has been a month and more than ten commits had been made. I think it is almost time for the next release.
However, does anyone have reason not to include this in the next release? So, I'll merge this

**Pull request merged**